### PR TITLE
System Messages Fix

### PR DESCRIPTION
**Pull request by MatkoTom**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant
motivation and context. List any dependencies that are required for this change.

Various bugfixes tied to the system messsages, like being able to swipe them and reply to them etc. Also fixed a critical issue when users are leaving rooms. It seems that we removed all user data from junction tables. 

Also, updated the System messages UI and handled logic with displaying usernames and images.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can
reproduce. Please also list any relevant details for your test configuration. All testing was done
manually by following, plainly, functionality requirements. If there are some special edge cases or
special ways that things need to be tested they will be mentioned below.

Dec testing

**Test Configuration**:

* Firmware version:
* Hardware: Samsung Galaxy S22
* SDK: Android 14

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/src/main/java/com/clover/studio/spikamessenger/utils/MessageSwipeHelper.kt [hunk 1]
```diff
@@ -59,6 +59,7 @@ class MessageSwipeController(
 
         return if (messageRecords[viewHolder.absoluteAdapterPosition].message.deleted == null
             || messageRecords[viewHolder.absoluteAdapterPosition].message.deleted == true
+            || messageRecords[viewHolder.absoluteAdapterPosition].message.type == Const.JsonFields.SYSTEM_TYPE
```
Comment (ibencetic-CS) [inline app/src/main/java/com/clover/studio/spikamessenger/utils/MessageSwipeHelper.kt:62]: Maybe add a method `canDelete` in message class to move these checks from the helper.

**Review by ibencetic-CS — DISMISSED** [review]

**Review by ibencetic-CS — DISMISSED** [review]
